src/Ideal.jl

```diff
@@ -4,15 +4,31 @@
 #
 ###############################################################################
 
-function ideal
+# We assume that the functions
+#   ideal(R::Ring, x::RingElement)
+#   ideal(R::Ring, xs::AbstractVector)
+# are implemented by anyone implementing ideals for AbstractAlgebra rings.
+# The functions in this file extend the interface for `ideal`.
+
+function *(R::Ring, x::Any)
+  return ideal(R, R(x))
```

Comment:
Since you now provide `ideal(R::Ring, x::Any)` you can simplify here again to this:
```suggestion
  return ideal(R, x)
```
the advantage is that if some ring type wants to implement say `ideal(R::MyRing, n::Int)` in a special optimized fashion that avoids constructing `R(n)`, then `R*2` would benefit from that